haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
+
+
+class SentenceTransformersEmbeddingBackend:
+    """
+    Singleton class to manage SentenceTransformers embeddings.
+    """
+
+    _instances: Dict[str, "SentenceTransformersEmbeddingBackend"] = {}
+
+    def __new__(cls, *args, **kwargs):
+        args_kwargs_str = str(args) + str(kwargs)
+        instance_id = hashlib.md5(args_kwargs_str.encode())
+
+        if instance_id in cls._instances:
+            return cls._instances[instance_id]
+
+        instance = super().__new__(cls)
+        cls._instances[instance_id] = instance
+        return instance
+
+    def __init__(
+        self, model_name_or_path: str, device: Optional[str] = None, use_auth_token: Union[bool, str, None] = None
+    ):
+        sentence_transformers_import.check()
+        if not hasattr(self, "model"):
```

Comment:
Are there situations in which `model` is already set? :eyes: I can't see that. Maybe it's a leftover

Reply:
That happens cause we reuse the instances.

Probably we can skip `__init__` completely if `self.model` is set.